### PR TITLE
Simplify the way "edit list" title is determined.

### PR DESCRIPTION
**Pull request by demiankatz**
This addresses a problem discovered while testing #2417: in PostgreSQL, new rows fetched from the database come with the ID pre-populated from the sequence, even if they are never saved. This interferes with correctly differentiating between "create row" and "edit row" titles. Fortunately, an existing flag offers an easier way to do this correctly.

Note that the underlying cause of the problem may still be a problem, since hitting "create list" repeatedly exhausts IDs unnecessarily. However, I'm not going to worry too much about that right now, since I'm hopeful that switching everything to Doctrine will likely change the situation anyway (see #2233).

TODO
- [x] Run full test suite with MySQL
- [x] Run full test suite with PostgreSQL

### CODE REVIEW

**Review by EreMaijala — APPROVED** [review]

**Review by EreMaijala — CHANGES_REQUESTED** [review]
The change is good, but should https://github.com/vufind-org/vufind/blob/ea04a81dcd215a053f1ae1af57984e7377533402/themes/bootstrap3/templates/myresearch/editlist.phtml#L22 be changed as well?

**Comment by demiankatz** [comment]
Good catch, @EreMaijala -- I've made the suggested edit and re-run tests in both MySQL and PostgreSQL. I think all is well now, but I'll await your approval before merging.

**Review by EreMaijala — APPROVED** [review]

**Pull request merged**
